packages/react/src/queryKeeper.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
+
+interface useQueryKeeperProps {
+  routes: {
+    [key: string]: string;
+  };
+}
+
+const removeRedirectUrlFromQuery = (query: string) => {
+  const q = new URLSearchParams(query);
+  q.delete('redirectUrl');
+  const cleanedQuery = q.toString();
+
+  return cleanedQuery;
+};
+
+export const useQueryKeeper = ({ routes }: useQueryKeeperProps): void => {
+  const queryParams = useRef<string>();
+  const prevPathname = useRef<string>();
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (!!history.location.search) {
+      queryParams.current = history.location.search;
+      prevPathname.current = history.location.pathname;
+    }
+  }, []);
+
+  history.listen((listener) => {
+    if (listener.search) {
+      queryParams.current = listener.search;
```

Comment:
what is the difference between the use case in this line to link 26?

Reply:
no need